src/test/ui/consts/const-eval/validate_uninhabited_zsts.rs

```diff
@@ -6,16 +6,21 @@ const fn foo() -> ! {
     //~| WARN the type `!` does not permit zero-initialization [invalid_value]
 }
 
-#[derive(Clone, Copy)]
-enum Empty { }
+pub mod empty {
```

Comment:
```suggestion
// Type defined in a submodule, so that it is not "visibly"
// uninhabited (which would change interpreter behavior).
pub mod empty {
```